src/Automation/SnapshotCommand.cs

```diff
@@ -18,8 +18,9 @@ static class SnapshotCommand
         /// <param name="config">A set of configuration options</param>
         /// <param name="scanTools">A set of tools for writing output files,
         /// creating the expected results format, and finding the target element to scan</param>
+        /// <param name="outputFileNameWithoutExtension">The name (without extension) for the output file (can be null)</param>
         /// <returns>A SnapshotCommandResult that describes the result of the command</returns>
-        public static ScanResults Execute(Config config, IScanTools scanTools)
+        public static ScanResults Execute(Config config, IScanTools scanTools, string outputFileNameWithoutExtension = null)
```

Comment:
This file doesn't need to be touched at all. You can set the given file name in the Scanner.Scan code before calling Execute.

Reply:
I considered setting the file name in `Scanner.Scan`, but it's not guaranteed to work if multiple threads call `Scanner.Scan` concurrently and pass in different file names. The `OutputFileHelper` object persists for the lifetime of the scanner, and is shared by all scans. Since `Scanner.Scan` provides no serialization, a race condition would be created--it would be a "last writer wins" scenario, but it's indeterminate which thread would be the last writer.

Serialization is already provided by `ExecutionWrapper.ExecuteCommand`, so by setting the output file _inside_ the lambda, we completely remove the race condition, no matter how many threads are involved.